staging/src/k8s.io/legacy-cloud-providers/aws/aws.go

```diff
@@ -3521,9 +3524,10 @@ func (c *Cloud) EnsureLoadBalancer(ctx context.Context, clusterName string, apiS
 
 	sslPorts := getPortSets(annotations[ServiceAnnotationLoadBalancerSSLPorts])
 	for _, port := range apiService.Spec.Ports {
-		if port.Protocol != v1.ProtocolTCP {
-			return nil, fmt.Errorf("Only TCP LoadBalancer is supported for AWS ELB")
+		if err := checkProtocol(port.Protocol, annotations); err != nil {
+			return nil, err
 		}
+
```

Comment:
I think you also need to think about HealthCheckProtocol, and also what about when when we are listening on port udp & tcp on the same port - I think we need to worry about the protocol + port now in the primary key of maps e.g. https://github.com/kubernetes/kubernetes/blob/c7f9dd0bafe2f3328148a85eed9c18322b9f308e/staging/src/k8s.io/legacy-cloud-providers/aws/aws_loadbalancer.go#L189